tests/test_coils.py

```diff
@@ -24,51 +26,110 @@ class TestCoil:
     @pytest.mark.unit
     def test_biot_savart_all_coils(self):
         """Test biot-savart implementation against analytic formula."""
+        coil_grid = LinearGrid(zeta=100, endpoint=False)
+
         R = 2
         y = 1
         I = 1e7
+
         By_true = 1e-7 * 2 * np.pi * R**2 * I / (y**2 + R**2) ** (3 / 2)
-        B_true = np.array([0, By_true, 0])
+        Bz_true = 1e-7 * 2 * np.pi * R**2 * I / (y**2 + R**2) ** (3 / 2)
+
+        B_true_xyz = np.atleast_2d([0, By_true, 0])
+        grid_xyz = Grid([10, y, 0])
+        grid_rpz = Grid(xyz2rpz(grid_xyz.nodes))
+        B_true_rpz_xy = xyz2rpz_vec(
+            B_true_xyz, x=grid_xyz.nodes[:, 0], y=grid_xyz.nodes[:, 1]
+        )
+        B_true_rpz_phi = xyz2rpz_vec(B_true_xyz, phi=grid_rpz.nodes[:, 1])
 
         # FourierXYZCoil
         coil = FourierXYZCoil(I)
-        grid = LinearGrid(zeta=100, endpoint=False)
-        B_approx = coil.compute_magnetic_field(
-            Grid([[10, y, 0], [10, -y, 0]]), basis="xyz", grid=grid
-        )[0]
+        B_xyz = coil.compute_magnetic_field(
+            grid_xyz, basis="xyz", source_grid=coil_grid
+        )
+        B_rpz = coil.compute_magnetic_field(
+            grid_rpz, basis="rpz", source_grid=coil_grid
+        )
```

Comment:
I realized we didn't have any test to check that `compute_magnetic_field(basis="rpz")` gives the same result as `compute_magnetic_field(basis="xyz")`, so I added it here for each coil.